host-spec/c06-consensus.tm

```diff
@@ -337,6 +423,25 @@
     there is no ambiguity about the underlying node.
   </definition>
 
+  At the end of each sync-epoch (def. <reference|defn-sync-epoch>) these
+  block arrivel times are used to update the current clock offset using the
+  median algorithm (alg. <reference|algo-slot-time>).
+
+  <\definition>
+    <label|defn-sync-epoch>A <with|font-series|bold|sync epoch> (unrelated to
+    the epoch defined in <reference|defn-epoch-slot>) is the inteval at which
+    each validator (re-)evaluates its local clock offsets. The first
+    sync-epoch <math|\<frak-E\><rsub|1>> starts just after the genesis block
+    is released. The other sync-epochs <math|\<frak-E\><rsub|i>> start when
+    the slot number of the last (probabilistically) finalized block is
+    <math|<wide|s|\<bar\>><rsub|i>> which is the smallest slot number such
+    that <math|<wide|s|\<bar\>><rsub|i><rsub|>-<wide|s|\<bar\>><rsub|i-1>\<geq\>s<rsub|cq>>
+    where <math|<wide|s|\<bar\>><rsub|i-1>> is the slot number of the last
+    (probabilistically) finalized block in the previous sync-epoch
+    <math|\<frak-E\><rsub|i-1>>. If the previous epoch is the first epoch
+    <math|\<frak-E\><rsub|1>> then <math|<wide|s|\<bar\>><rsub|i-1>=<wide|s|\<bar\>><rsub|i>=0>.
+  </definition>
+
   In addition to the arrival time of block <math|B>, the block producer also
   needs to know how many slots have passed since the arrival of <math|B>.
   This value is formalized in Definition <reference|defn-slot-offset>.
```

Comment:
Why don't we give a formula for the slot-offset? I guess it should be If k < l, (l - k- 1) * eplen + eplen - i + j, if k = l, j - i

Reply:
Yes, this is another reason to simplify this definition.